File: LeetCode-Java/p11.java
class Solution {
    public int maxArea(int[] height) {
        int p = 0;
        int q = height.length - 1;
        int maxArea = 0;

        while (p < q) {
            // Calculate the area with the current two pointers
            int area = Math.min(height[p], height[q]) * (q - p);
            maxArea = Math.max(maxArea, area);

            // Update the pointer with the shorter height
            if (height[p] < height[q]) {
                p++;
            } else {
                q--;
            }
        }

        return maxArea;
    }
}

File: LeetCode/p11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        p, q = 0, len(height) - 1
        max_area = 0
        while p < q:
            max_area = max(max_area, min(height[p], height[q]) * (q - p))
            if height[p] < height[q]:
                p += 1
            else:
                q -= 1

        return max_area